parsons/bill_com/bill_com.py

```diff
@@ -171,80 +202,110 @@ def checkCustomer(self, customer1, customer2):
+                if self.check_customer(customer, existing_customer):
+                    return existing_customer
+            # customer doesn't exist, create
+            data = {
+                "obj": customer
+            }
+            return self.get_request_response(data, "Create", "Customer")
 
-    def createInvoice(self, invoice):
+    def create_invoice(self, customer_id, invoice_number, invoice_date, due_date, invoice_line_items, **kwargs):
         """
         `Args:`
-            invoice: dict
-                A dictionary of data on the invoice
+            customer_id: str
+                The customer's id
+            invoice_number: str
+                The invoice number. Every invoice must have a distinct invoice number.
+            invoice_date: str
+                The invoice date. This can be the date the invoice was generated of any other relevant date.
+            due_date: str
+                The date on which the invoice is due.
+            invoice_line_items: list
+                A list of dicts, one for each line item in the invoice. The only required field is "quantity".
+            **kwargs:
+                Any other invoice details to pass.
 
         `Returns:`
             A dictionary of the invoice's information including an id.
         """
+        for invoice_line_item in invoice_line_items:
+            if "entity" not in invoice_line_item:
+                invoice_line_item['entity'] = 'InvoiceLineItem'
         data = {
-            "obj": invoice
+            "obj": {"customerId": customer_id,
+                    "invoiceNumber": invoice_number,
+                    "invoiceDate": invoice_date,
+                    "dueDate": due_date,
+                    "invoiceLineItems": invoice_line_items,
+                    **locals()['kwargs']
+            }
         }
-        return self.getRequestResponse(data, "Create", "Invoice")
+        return self.get_request_response(data, "Create", "Invoice")
 
-    def sendInvoice(self, invoiceId, fromUserId, toEmailAddresses, messageSubject, messageBody):
+    def send_invoice(self, invoice_id, from_user_id, to_email_addresses, message_subject, message_body, **kwargs):
         """
         `Args:`
-            invoiceId: str
+            invoice_id: str
                 The id of the invoice to send
-            fromUserId: str
+            from_user_id: str
                 The id of the Bill.com user from whom to send the email
-            toEmailAddresses:
+            to_email_addresses:
                 The customer's email address
-            messageSubject:
+            message_subject:
                 The subject of the email to send to the customer
-            messageBody:
-                The boyd of the email to send to the customer
+            message_body:
+                The body of the email to send to the customer
+            **kwargs:
+                Any other details for sending the invoice
```

Comment:
I love that you've added these to all the methods, but I don't actually see where these are being incorporated in the code. It's possible we'd never need to, so maybe we can just take this out for now until the need arises.